src/TLuaInterpreter.cpp

```diff
@@ -1787,10 +1789,27 @@ int TLuaInterpreter::saveMap(lua_State* L)
         }
     }
 
+    if (lua_gettop(L) == 2) {
+        if (!lua_isstring(L, 1)) {
+            lua_pushstring(L, "saveMap: wrong argument location.");
```

Comment:
If you're up for it, migrating saveMap to the better error msg style (blah expected, got whatever) would be cool.